packages/search/tests/unit/services/navi-search/navi-report-and-dashboard-search-provider-test.js

```diff
@@ -0,0 +1,105 @@
+    const store = this.owner.lookup('service:store'),
+      mockAuthor = store.createRecord('user', { id: 'ciela' });
+    this.owner.register(
+      'service:user',
+      Service.extend({
+        getUser: () => mockAuthor
+      })
+    );
+  });
+
+  test('it exists', function(assert) {
+    const service = this.owner.lookup('service:navi-search/navi-report-and-dashboard-search-provider');
+    assert.ok(service);
+  });
+
+  test('construct full search query for reports', function(assert) {
+    assert.deepEqual(
+      this.service._constructSearchQuery({ title: 'Hyrule', request: 'clicks' }, 'navi_user', 'reports'),
+      {
+        filter: { reports: '(title==*Hyrule*,request==*clicks*);author==*navi_user*' }
+      },
+      'Report constructs the correct query for the api with both filter parameters and author.'
+    );
+  });
+
+  test('construct only query parameters search query for reports', function(assert) {
+    assert.deepEqual(
+      this.service._constructSearchQuery({ title: 'Hyrule', request: 'clicks' }, null, 'reports'),
+      {
+        filter: { reports: '(title==*Hyrule*,request==*clicks*)' }
+      },
+      'Report constructs the correct query for the api with filter parameters.'
+    );
+  });
+
+  test('construct only author search query for reports', function(assert) {
+    assert.deepEqual(
+      this.service._constructSearchQuery(null, 'navi_user', 'reports'),
+      {
+        filter: { reports: 'author==*navi_user*' }
+      },
+      'Report constructs the correct query for the api with author.'
+    );
+  });
+
+  test('construct full search query for dashboards', function(assert) {
+    assert.deepEqual(
+      this.service._constructSearchQuery({ title: 'Hyrule' }, 'navi_user', 'dashboards'),
+      {
+        filter: { dashboards: '(title==*Hyrule*);author==*navi_user*' }
+      },
+      'Report constructs the correct query for the api with both filter parameters and author.'
+    );
+  });
+
+  test('construct only query parameters search query for dashboards', function(assert) {
+    assert.deepEqual(
+      this.service._constructSearchQuery({ title: 'Hyrule' }, null, 'dashboards'),
+      {
+        filter: { dashboards: '(title==*Hyrule*)' }
+      },
+      'Report constructs the correct query for the api with filter parameters.'
+    );
+  });
+
+  test('construct only author search query for dashboards', function(assert) {
+    assert.deepEqual(
+      this.service._constructSearchQuery(null, 'navi_user', 'dashboards'),
+      {
+        filter: { dashboards: 'author==*navi_user*' }
+      },
+      'Report constructs the correct query for the api with author.'
+    );
+  });
+
+  test('search by user search returns reports and dashboards', async function(assert) {
+    const results = await this.service.search('Revenue');
+    const author = await results[0].author;
+    assert.ok(results[0].title.includes('Revenue'), 'The service returns a report that includes the requested title.');
```

Comment:
Can we assert against all results rather than just the first one? We want to make sure that all results have `Revenue` in their title field and have author `ciela`